server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreFunction.java

```diff
@@ -22,6 +22,8 @@
 
 public class ScriptScoreFunction extends ScoreFunction {
 
+    private static final ScoreScript.ExplanationHolder DUMMY_EXPLAIN_HOLDER = new ScoreScript.ExplanationHolder();
```

Comment:
underlying folks should check for null. Passing this in implies that explain is occurring. It might be the script's plugin explanation takes compute power, we need to be able to signal that explain actually isn't occurring.

Reply:
So, we shouldn't have this dummy holder.